### PR TITLE
fix: When executing cli.setuphijackconn, call cli.dialer () after exe…

### PR DESCRIPTION
**Pull request by yunnet**
…cuting "dialer := cli.dialer ()",

TLSClientConfig is not null when Scheme =" HTTP ", causing an error.
The error message is: OCI runtime exec failed: exec failed: unable to start container process: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?

- addresses https://github.com/moby/moby/issues/43783

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
Thanks, but please sign the commit for DCO
https://github.com/apps/dco

(run `git commit -a -s --amend`, and make sure that the `Signed-off-by: NAME <EMAIL>` line with your real name is included in the commit message)

File: client/client.go [hunk 1]
```diff
@@ -311,6 +311,9 @@ func ParseHostURL(host string) (*url.URL, error) {
 func (cli *Client) Dialer() func(context.Context) (net.Conn, error) {
 	return func(ctx context.Context) (net.Conn, error) {
 		if transport, ok := cli.client.Transport.(*http.Transport); ok {
+			if cli.scheme == "http" {
+				transport.TLSClientConfig = nil
+			}
```
Comment (thaJeztah) [inline client/client.go:316]: @AkihiroSuda I'm wondering what the root cause is of this; I had a quick look at some of the code, and looking at https://github.com/moby/moby/blob/0910306bf970603ce787466a98e4294ba81af841/client/client.go#L146-L158


That appears to change `scheme` to `https` if TLS Config was found; is something wrong with the initialisation code?

**Comment by thaJeztah** [comment]
Thanks for looking; let me close this one 👍

**Pull request closed without merging**